src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import { Button, Menu, Pagination, Typography } from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ClearIcon from '@mui/icons-material/Clear';
+import { createClient } from '@supabase/supabase-js';
+import AddIcon from '@mui/icons-material/Add';
+
+type InventoryItem = {
+  id: number;
+  item: string;
+  type: string;
+  quantity: number;
+  category: string;
+}
+
+
+const Inventory = () => {
+
+  const [originalData, setOriginalData] = useState<InventoryItem[]>([]);
+  const [data, setData] = useState<InventoryItem[]>([]);
+  const [type, setType] = useState('');
+  const [category, setCategory] = useState('');
+  const [status, setStatus] = useState('');
+  const [search, setSearch] = useState('');
+  const [anchorType, setAnchorType] = useState<null | HTMLElement>(null);
+  const [anchorCategory, setAnchorCategory] = useState<null | HTMLElement>(null);
+  const [anchorStatus, setAnchorStatus] = useState<null | HTMLElement>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 10;
+
+  const indexOfLastItem = currentPage * itemsPerPage;
+  const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+  const currentItems = data.slice(indexOfFirstItem, indexOfLastItem);
+
+  const handleTypeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorType(event.currentTarget);
+  };
+  const handleCategoryClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorCategory(event.currentTarget);
+  };
+  const handleStatusClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorStatus(event.currentTarget);
+  };
+
+  const handleTypeClose = () => {
+    setAnchorType(null);
+  }
+
+  const handleMenuTypeClick = (value: string) => {
+    setType(value)
+    handleFilter();
+    handleTypeClose();
+  }
+
+  const handleCategoryClose = () => {
+    setAnchorCategory(null);
+  }
+
+  const handleMenuCategoryClick = (value: string) => {
+    setCategory(value);
+    handleFilter();
+    handleCategoryClose();
+  }
+
+  const handleStatusClose = () => {
+    setAnchorStatus(null);
+  }
+
+  const handleMenuStatusClick = (value: string) => {
+    setStatus(value);
+    handleFilter();
+    handleStatusClose();
+  }
+
+  const clearTypeFilter = () => {
+    setType('');
+  }
+  const clearCategoryFilter = () => {
+    setCategory('');
+  }
+  const clearStatusFilter = () => {
+    setStatus('');
+  }
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const handlePageChange = (_event: React.ChangeEvent<unknown>, value: number) => {
+    setCurrentPage(value);
+  }
+
+  const handleFilter = () => {
```

Comment:
To improve performance, Supabase can be used to filter on the database side based on certain conditions. Here's an example:

const { data: inventory, error } = await supabase
  .from('inventory')
  .select(`id, item, type, quantity, category`)
  .filter('category', 'eq', category) // filter by category
  .range(startIndex, endIndex);

Reply:
Since we're moving away from supabase, this can be handled when integrating Azure db.

Reply:
agreed, let's not fix supabase issues. Great suggestion though.